tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala

```diff
@@ -0,0 +1,237 @@
+
+  @Test
+  def getTokenShouldReturnJWTTokenResponse(): Unit = {
+    val longLivedToken: String = createLongLivedToken()
+
+    val response: String = `given`()
+      .header(buildBearerTokenHeader(longLivedToken))
+      .queryParam("type", "shortLived")
+      .queryParam("deviceId", DEVICE_ID)
+    .when()
+      .basePath(BASE_PATH)
+      .get()
+    .`then`()
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    assertThatJson(response)
+      .isEqualTo(
+        s"""
+          |{
+          |  "token" : "$${json-unit.ignore}",
+          |  "expiresOn" : "$${json-unit.ignore}"
+          |}""".stripMargin)
+  }
+
+  @ParameterizedTest
+  @MethodSource(value = Array("unSupportAuthenHeaderStream"))
+  def getTokenShouldNotAcceptOtherAuthenWithoutLongLivedToken(authenHeader: Header): Unit = {
+    val response: String = `given`()
+      .header(authenHeader)
+      .queryParam("type", "shortLived")
+      .queryParam("deviceId", DEVICE_ID)
+    .when()
+      .basePath(BASE_PATH)
+    .get()
+      .`then`()
+      .statusCode(HttpStatus.SC_UNAUTHORIZED)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    assertThatJson(response)
+      .isEqualTo(
+        s"""{
+           |    "type": "about:blank",
+           |    "status": 401,
+           |    "detail": "No valid authentication methods provided"
+           |}""".stripMargin)
+  }
+
+  @Test
+  def jwtTokenResponseShouldGrantAuthToJmapEndpoint(): Unit = {
+    val longLivedToken: String = createLongLivedToken()
+
+    val jwtToken: String = `given`()
+      .header(buildBearerTokenHeader(longLivedToken))
+      .queryParam("type", "shortLived")
+      .queryParam("deviceId", DEVICE_ID)
+    .when()
+      .basePath(BASE_PATH)
+      .get()
+    .`then`()
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract()
+      .jsonPath()
+      .get("token")
+
+    `given`
+      .basePath("/jmap")
+      .headers(getHeadersWith(new Header(AUTHORIZATION_HEADER, s"Bearer $jwtToken")))
+      .body(ECHO_REQUEST_OBJECT)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+  }
+
+  @Test
+  def getTokenShouldFailWhenMissingTypeParameter(): Unit = {
+    val longLivedToken: String = createLongLivedToken()
+
+    val response: String = `given`()
+      .header(buildBearerTokenHeader(longLivedToken))
+      .queryParam("deviceId", DEVICE_ID)
+    .when()
+      .basePath(BASE_PATH)
+      .get()
+    .`then`()
+      .statusCode(HttpStatus.SC_BAD_REQUEST)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    assertThatJson(response)
+      .isEqualTo(
+        s"""{
+           |    "type": "about:blank",
+           |    "status": 400,
+           |    "detail": "Invalid request: 'type' must be shortLived"
+           |}""".stripMargin)
+  }
+
+  @Test
+  def getTokenShouldFailWhenMissingDeviceIdParameter(): Unit = {
```

Comment:
Did you tested what happens with a bad deviceId?

Reply:
it's in is todo list https://github.com/linagora/tmail-backend/pull/234#issuecomment-923866977, thus why I didn't comment this :)